changelog.d/9145.bugfix

```diff
@@ -0,0 +1 @@
+Fix "UnboundLocalError: local variable 'length' referenced before assignment" errors when the response body exceeds the expected size. This bug was introduced in v1.25.0.
```

Comment:
I think the error here is the same in both cases, although the stack would be different.